tests/test_result.py

```diff
@@ -77,29 +77,20 @@ def test_get_commandlog(pkg):
 @pytest.mark.parametrize('os_name', ['posix', 'nt'])
 def test_invocation_error(exit_code, os_name, mocker, monkeypatch):
     monkeypatch.setattr(os, 'name', value=os_name)
-    mocker.spy(tox, '_exit_code_str')
+    mocker.spy(tox.exception, 'exit_code_str')
+    result = str(tox.exception.InvocationError("<command>", exit_code=exit_code))
+    # check that mocker works, because it will be our only test in
+    # test_z_cmdline.py::test_exit_code needs the mocker.spy above
+    assert tox.exception.exit_code_str.call_count == 1
+    assert tox.exception.exit_code_str.call_args == mocker.call(
+        'InvocationError', "<command>", exit_code)
     if exit_code is None:
-        exception = tox.exception.InvocationError("<command>")
+        assert "(exited with code" not in result
     else:
-        exception = tox.exception.InvocationError("<command>", exit_code)
-    result = str(exception)
-    # check that mocker works,
-    # because it will be our only test in test_z_cmdline.py::test_exit_code
-    # need the mocker.spy above
-    assert tox._exit_code_str.call_count == 1
-    assert tox._exit_code_str.call_args == mocker.call('InvocationError', "<command>", exit_code)
-    if exit_code is None:
-        needle = "(exited with code"
-        assert needle not in result
-    else:
-        needle = "(exited with code %d)" % exit_code
-        assert needle in result
-        note = ("Note: this might indicate a fatal error signal")
+        assert "(exited with code %d)" % exit_code in result
+        note = "Note: this might indicate a fatal error signal"
         if (os_name == 'posix') and (exit_code == 128 + signal.SIGTERM):
             assert note in result
-            number = signal.SIGTERM
-            name = "SIGTERM"
-            signal_str = "({} - 128 = {}: {})".format(exit_code, number, name)
-            assert signal_str in result
+            assert "(%s - 128 = %d: SIGTERM)" % (exit_code, signal.SIGTERM) in result
```

Comment:
this is hard to read wit the % notation; e.g. is ``(exit_code, signal.SIGTERM) in result`` or ``((%s - 128 = %d: SIGTERM)" % (exit_code, signal.SIGTERM)) in result``